pkg/proxy/iptables/proxier.go

```diff
@@ -1174,15 +1174,17 @@ func (proxier *Proxier) syncProxyRules() {
 					// This currently works for loadbalancers that preserves source ips.
 					// For loadbalancers which direct traffic to service NodePort, the firewall rules will not apply.
 
-					args = append(args[:0],
-						"-A", string(kubeServicesChain),
-						"-m", "comment", "--comment", fmt.Sprintf(`"%s loadbalancer IP"`, svcNameString),
-						"-m", protocol, "-p", protocol,
-						"-d", utilproxy.ToCIDR(net.ParseIP(ingress)),
-						"--dport", strconv.Itoa(svcInfo.Port()),
-					)
-					// jump to service firewall chain
-					writeLine(proxier.natRules, append(args, "-j", string(fwChain))...)
+					if !utilfeature.DefaultFeatureGate.Enabled(features.LoadBalancerIPMode) || *ingress.IPMode == v1.LoadBalancerIPModeVIP {
```

Comment:
How about capturing this as a helper function like:

```
func isVIP(ing *LoadBalancerIngress) bool {
    if !utilfeature.DefaultFeatureGate.Enabled(features.LoadBalancerIPMode) {
        return true // backwards compat
    }
    return ing.IPMode == v1.LoadBalancerIPModeVIP
}
```

This makes the eventual cleanup easier.  We should probably be doing more of this.